backend/src/packages/orders/order.controller.ts

```diff
@@ -0,0 +1,616 @@
+ *             $ref: '#/components/schemas/Order/properties/scheduledTime'
+ *           carsQty:
+ *             $ref: '#/components/schemas/Order/properties/carsQty'
+ *           startPoint:
+ *             $ref: '#/components/schemas/Order/properties/startPoint'
+ *           endPoint:
+ *             $ref: '#/components/schemas/Order/properties/endPoint'
+ *           driverId:
+ *             $ref: '#/components/schemas/Order/properties/driverId'
+ *       CreateOrderWithNotRegisteredUser:
+ *         type: object
+ *         required:
+ *         - scheduledTime
+ *         - carsQty
+ *         - startPoint
+ *         - endPoint
+ *         - driverId
+ *         - customerName
+ *         - customerPhone
+ *         properties:
+ *           scheduledTime:
+ *             $ref: '#/components/schemas/Order/properties/scheduledTime'
+ *           carsQty:
+ *             $ref: '#/components/schemas/Order/properties/carsQty'
+ *           startPoint:
+ *             $ref: '#/components/schemas/Order/properties/startPoint'
+ *           endPoint:
+ *             $ref: '#/components/schemas/Order/properties/endPoint'
+ *           driverId:
+ *             $ref: '#/components/schemas/Order/properties/driverId'
+ *           customerName:
+ *             $ref: '#/components/schemas/Order/properties/customerName'
+ *           customerPhone:
+ *             $ref: '#/components/schemas/Order/properties/customerPhone'
+ *     OrderCreationError:
+ *       allOf:
+ *         - $ref: '#/components/schemas/ErrorType'
+ *         - type: object
+ *           properties:
+ *             message:
+ *               type: string
+ *               enum:
+ *                 - Order can't be created!
+ *     DriverNotExistError:
+ *       allOf:
+ *         - $ref: '#/components/schemas/ErrorType'
+ *         - type: object
+ *           properties:
+ *             message:
+ *               type: string
+ *               enum:
+ *                 - Driver does not exist!
+ *     BusinessNotExistError:
+ *       allOf:
+ *         - $ref: '#/components/schemas/ErrorType'
+ *         - type: object
+ *           properties:
+ *             message:
+ *               type: string
+ *               enum:
+ *                 - Business does not exist!
+ *
+ */
+
+class OrderController extends Controller {
+  private orderService: OrderService;
+
+  private businessService: BusinessService;
+
+  private driverService: DriverService;
+
+  public constructor({
+    logger,
+    orderService,
+    businessService,
+    driverService,
+  }: {
+    logger: ILogger;
+    orderService: OrderService;
+    businessService: BusinessService;
+    driverService: DriverService;
+  }) {
+    super(logger, ApiPath.ORDERS);
+
+    this.orderService = orderService;
+
+    this.businessService = businessService;
+
+    this.driverService = driverService;
+
+    this.addRoute({
+      path: OrdersApiPath.ROOT,
+      method: 'GET',
+      authStrategy: AuthStrategy.VERIFY_JWT,
```

Comment:
this route should be available only for business, is it now implemented such way?

Reply:
restricted to business